spec/index.md

```diff
@@ -5921,6 +5930,12 @@ The pseudo-code below does *not* explicitly enforce the restrictions of which im
       else
         Trap {cycles_used = es.cycles_used;}
 
+    ic0.in_replicated_execution<es>() : i32 =
+      if es.context = s then Trap {cycles_used = es.cycles_used;}
+      if es.params.sysenv.certificate = NoCertificate
```

Comment:
Do we want to specify the return value based on the certificate or use `es.context` instead?

Reply:
Good question. I thought doing it this way is future proof since we won't change how certified state is created/handled, but using es would be the clean way to do it (with the caveat that we may forget to update it if/when we introduce some new execution modes). wdyt?

Reply:
Either way sounds good to me if you think it is future proof.